uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -4253,6 +4223,188 @@ public function testTwoLeaveRequestsOneAMAndAnotherPMCanBeCreatedForTheSameDay()
     $this->assertNotNull($leaveRequest2->id);
   }
 
+  public function testTwoTOILRequestsCanBeCreatedForTheSameDay() {
+    $date = '2018-04-13';
+    $contactId = 1;
+    $absenceType = AbsenceTypeFabricator::fabricate([
+      'allow_accruals_request' => true,
+      'allow_accrue_in_the_past' => true,
+      'calculation_unit' => 2
+    ]);
+
+    $absencePeriod = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2018-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2018-12-31'),
+    ]);
+
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $absenceType->id,
+      'contact_id' => $contactId,
+      'period_id' => $absencePeriod->id
+    ]);
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contactId],
+      ['period_start_date' => CRM_Utils_Date::processDate('2018-01-01')]
+    );
+
+    $this->createLeaveBalanceChange($periodEntitlement->id, 100);
+
+    $params = [
+      'type_id' => $absenceType->id,
+      'contact_id' => $contactId,
+      'from_date' => CRM_Utils_Date::processDate($date . ' 10:00:00'),
+      'to_date' => CRM_Utils_Date::processDate($date . ' 10:15:00'),
+      'request_type' => LeaveRequest::REQUEST_TYPE_TOIL,
+      'toil_to_accrue' => 2
+    ];
+
+    $leaveRequest1 = LeaveRequestFabricator::fabricate($params, true);
+
+    $params['from_date'] = CRM_Utils_Date::processDate($date . ' 11:00:00');
+    $params['to_date'] = CRM_Utils_Date::processDate($date . ' 11:15:00');
+
+    $leaveRequest2 = LeaveRequestFabricator::fabricate($params, true);
+
+    $this->assertNotNull($leaveRequest1->id);
+    $this->assertNotNull($leaveRequest2->id);
+  }
+
+  public function testTwoLeaveRequestsBothForAMCannotBeCreatedForTheSameDay() {
+    $date = '2018-04-13';
+    $contactId = 1;
+    $absenceType = AbsenceTypeFabricator::fabricate();
+    $halfDayAMID = $this->leaveRequestDayTypes['half_day_am']['value'];
+
+    $absencePeriod = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2018-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2018-12-31'),
+    ]);
+
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $absenceType->id,
+      'contact_id' => $contactId,
+      'period_id' => $absencePeriod->id
+    ]);
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contactId],
+      ['period_start_date' => CRM_Utils_Date::processDate('2018-01-01')]
+    );
+
+    $this->createLeaveBalanceChange($periodEntitlement->id, 100);
+
+    $this->setExpectedException(
+      'CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException',
+      'This leave request overlaps with another request. Please modify dates of this request'
+    );
+
+    for ($i = 1; $i <= 2; $i++) {
```

Comment:
Can you copy this twice rather than use a loop? store the parameters in a variable and use it twice for the fabricators .

Reply:
Done.